go/vt/vtorc/logic/tablet_discovery.go

```diff
@@ -64,16 +65,24 @@ func RegisterFlags(fs *pflag.FlagSet) {
 // OpenTabletDiscovery opens the vitess topo if enables and returns a ticker
 // channel for polling.
 func OpenTabletDiscovery() <-chan time.Time {
-	// TODO(sougou): If there's a shutdown signal, we have to close the topo.
 	ts = topo.Open()
 	tmc = tmclient.NewTabletManagerClient()
 	// Clear existing cache and perform a new refresh.
 	if _, err := db.ExecVTOrc("delete from vitess_tablet"); err != nil {
 		log.Error(err)
 	}
+	// We refresh all information from the topo once before we start the ticks to do it on an timer.
+	populateAllInformation()
 	return time.Tick(time.Second * time.Duration(config.Config.TopoInformationRefreshSeconds)) //nolint SA1015: using time.Tick leaks the underlying ticker
 }
 
+// populateAllInformation initializes all the information for VTOrc to function.
+func populateAllInformation() {
+	refreshAllInformation()
+	// We have completed one discovery cycle in the entirety of it. We should update the process health.
```

Comment:
```suggestion
	// We have completed one full discovery cycle. We should update the process health.
```

Reply:
Do we even need to say full/entire at all? Is there such a thing as a partial discovery cycle?

Reply:
Yes, we do tablet discoveries, and shard record updates. So I think having full there makes sense, given that we're initializing eveything for the first time.